File: app/src/main/java/com/example/android/popularmovies/adapters/TrailersAdapter.java
package com.example.android.popularmovies.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.android.popularmovies.R;

public class TrailersAdapter extends RecyclerView.Adapter<TrailersAdapter.TrailersAdapterTrailerHolder> {

    private String[] trailers;

    private final TrailerClickListener trailerClickListener;
    public interface TrailerClickListener{
        void onTrailerClicked(String trailer);
    }

    public TrailersAdapter(TrailerClickListener trailerClickListener){
        this.trailerClickListener = trailerClickListener;
    }

    private Context context;
    @NonNull
    @Override
    public TrailersAdapterTrailerHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        View view = layoutInflater.inflate(R.layout.trailers_list_item, parent, false);
        return new TrailersAdapterTrailerHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TrailersAdapterTrailerHolder holder, int position) {
        String text = context.getString(R.string.trailer_value, (position + 1));
        holder.trailerText.setText(text);
    }

    @Override
    public int getItemCount() {
        if(trailers != null){
            return trailers.length;
        }
        return 0;
    }

    public void setTrailers(String[] trailers) {
        this.trailers = trailers;
    }

    class TrailersAdapterTrailerHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        final TextView trailerText;

        TrailersAdapterTrailerHolder(@NonNull View itemView) {
            super(itemView);
            trailerText = itemView.findViewById(R.id.trailer_label);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            int adapterPosition = getAdapterPosition();
            trailerClickListener.onTrailerClicked(trailers[adapterPosition]);

        }
    }

}

File: app/src/main/java/com/example/android/popularmovies/adapters/TrailersAdapter.kt
package com.example.android.popularmovies.adapters

import android.content.Context
import com.example.android.popularmovies.adapters.TrailersAdapter.TrailerClickListener
import androidx.recyclerview.widget.RecyclerView
import com.example.android.popularmovies.adapters.TrailersAdapter.TrailersAdapterTrailerHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.android.popularmovies.R
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import android.widget.TextView
import com.example.android.popularmovies.models.Video

class TrailersAdapter(private val trailers: List<Video>, private val trailerClickListener: TrailerClickListener) : RecyclerView.Adapter<TrailersAdapterTrailerHolder>() {

    interface TrailerClickListener {
        fun onTrailerClicked(trailer: Video)
    }

    private var context: Context? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TrailersAdapterTrailerHolder {
        context = parent.context
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(R.layout.trailers_list_item, parent, false)
        return TrailersAdapterTrailerHolder(view)
    }

    override fun onBindViewHolder(holder: TrailersAdapterTrailerHolder, position: Int) {
        val video = trailers[position]
        holder.trailerText.text = video.name
    }

    override fun getItemCount(): Int {
        return trailers.size
    }

    inner class TrailersAdapterTrailerHolder(itemView: View) : ViewHolder(itemView), View.OnClickListener {
        val trailerText: TextView = itemView.findViewById(R.id.trailer_label)

        override fun onClick(v: View) {
            val adapterPosition = adapterPosition
            trailerClickListener.onTrailerClicked(trailers[adapterPosition])
        }

        init {
            itemView.setOnClickListener(this)
        }
    }
}